pipelines/vagrant-linux/run.sh

```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+set -e -x
+
+# build deb packages
+
+# provision new kernel and start VM
```

Comment:
looks like comments on L4 and L6 are not needed.

In general, what happens if we re-run the pipeline? Does the kernel gets re-built again? Or is the pipeline reusing the previous gcc output and just recompiling what changed?

Also, if we generate new linux kernel packages, is vagrant going to load the new kernel? I believe that was the goal of L6 (add `vagrant reload` commands here). Looks like that's being done on the `deploy-kernel` stage, is that correct?